fontbro/font.py

```diff
@@ -216,15 +216,18 @@ def __init__(self, filepath, **kwargs):
         self._kwargs = None
         self._ttfont = None
 
-        if isinstance(filepath, (Path, str)):
-            self._init_with_filepath(str(filepath), **kwargs)
+        if isinstance(filepath, Path):
+            filepath = str(filepath)
+
+        if isinstance(filepath, str) or hasattr(filepath, "read"):
+            self._init_ttfont(filepath, **kwargs)
```

Comment:
If `filepath` is a `File` object, then the real filepath should be `filepath.name`, I think that splitting the two conditions would be logically more correct here.

Actually the `filepath` is stored in the `_filepath` variable to be used later when calling the `save` method, I'm curious to see what happens when saving a `Font` object instantiated using a `File` object.

Reply:
Not all file-object have a name, or name that actually exists in the filesystem. For example, a `BytesIO` object won't have a name.

Here's I'm passing the file handler directly to `TTFont`, which already has code to rewind the buffer (`.seek(0)`) and `.read()` it.

Reply:
> I'm curious to see what happens when saving a Font object instantiated using a File object.

if we call `.save(filepath)` with a valid `filepath` (ie: a `str`), it would just dump the file objects to that file.

What we need to decide if what behavior we should have when calling `.save()` with no arguments on a file-instantiated instance. Should we just return bytes? or a new `BytesIO`?

Reply:
> What we need to decide if what behavior we should have when calling .save() with no arguments on a file-instantiated instance. Should we just return bytes? or a new BytesIO?

Exactly. In this side case, probably the best thing to do is to raise an `Exception`, do you agree?

Reply:
Actually the `save` method is for saving the font to the file system only and I don't like the idea of changing the method output depending on the constructor input arguments.

Btw, if needed, I'm open to the idea of adding a `save_as_bytes` method.